File: 13.Linked_List/javascript/9.DeleteYandX.js
class Node {
    constructor(data) {
        this.data = data;
        this.next = null;
    }
}

function buildLinkedList(arr){
    if(arr.length ===0) return null
    
    let head = new Node(arr[0])
    let current = head
    
    for(let i=1;i<arr.length;i++){
        current.next = new Node(arr[i])
        current = current.next
    }
    return head
}

function deleteXAfterY(head, X, Y) {
    let current = head;

    while (current) {
        // Skip X nodes
        for (let i = 1; i < X && current != null; i++) {
            current = current.next;
        }

        // If end of list is reached
        if (current == null) break;

        // Start deleting Y nodes
        let temp = current.next;
        for (let i = 0; i < Y && temp != null; i++) {
            temp = temp.next;
        }

        // Connect current node to the (Y+1)th node
        current.next = temp;
        current = temp;
    }

    return head;
}

// Helper to print the linked list
function printList(head) {
    let result = [];
    let current = head;
    while (current) {
        result.push(current.data);
        current = current.next;
    }
    console.log(result.join(" "));
}

// Example usage:
let values = [2, 3, 2, 5, 3, 1];


let X = 3, Y = 2;
let head = buildLinkedList(values)
head = deleteXAfterY(head, X, Y);
printList(head); // Output: 2 3 2 1

File: 13.Linked_List/python/9.DeleteYandX.py
class Node: 
    def __init__(self, data): 
        self.data = data 
        self.next = None

def deleteYAfterX(head, N, X, Y):
    current = head
    
    while current:
       
        for i in range(1, X):
            if current is None:
                return head
            current = current.next

        if current is None:
            return head

        
        temp = current.next
        for i in range(Y):
            if temp is None:
                break
            temp = temp.next

       
        current.next = temp

        current = temp

    return head
